Source/Scene/PointCloudEyeDomeLighting.js

```diff
@@ -0,0 +1,400 @@
+        ShaderProgram,
+        Texture,
+        TextureMagnificationFilter,
+        TextureMinificationFilter,
+        TextureWrap,
+        VertexArray,
+        BlendEquation,
+        BlendFunction,
+        BlendingState,
+        StencilFunction,
+        StencilOperation,
+        PointCloudEyeDomeLightingShader
+    ) {
+    'use strict';
+
+    /**
+     * Eye dome lighting.
+     * Requires support for EXT_frag_depth, OES_texture_float, and WEBGL_draw_buffers extensions in WebGL 1.0.
+     *
+     * @private
+     */
+    function PointCloudEyeDomeLighting() {
+        this._framebuffers = undefined;
+        this._colorTexture = undefined; // color gbuffer
+        this._ecAndLogDepthTexture = undefined; // depth gbuffer
+        this._depthTexture = undefined; // needed to write depth so camera based on depth works
+        this._drawCommands = undefined;
+        this._clearCommands = undefined;
+
+        this._strength = 1.0;
+        this._radius = 1.0;
+    }
+
+    function addConstants(sourceFS, constantName, value) {
+        var finalSource = sourceFS;
+        if (typeof(value) === 'boolean') {
+            if (value !== false) {
+                finalSource = '#define ' + constantName + '\n' + sourceFS;
+            }
+        } else {
+            finalSource = '#define ' + constantName + ' ' + value + '\n' + sourceFS;
+        }
+        return finalSource;
+    }
+
+    function createSampler() {
+        return new Sampler({
+            wrapS : TextureWrap.CLAMP_TO_EDGE,
+            wrapT : TextureWrap.CLAMP_TO_EDGE,
+            minificationFilter : TextureMinificationFilter.NEAREST,
+            magnificationFilter : TextureMagnificationFilter.NEAREST
+        });
+    }
+
+    function destroyFramebuffers(processor) {
+        var framebuffers = processor._framebuffers;
+        if (!defined(framebuffers)) {
+            return;
+        }
+
+        processor._colorTexture.destroy();
+        processor._ecAndLogDepthTexture.destroy();
+        processor._depthTexture.destroy();
+        for (var name in framebuffers) {
+            if (framebuffers.hasOwnProperty(name)) {
+                framebuffers[name].destroy();
+            }
+        }
+
+        processor._framebuffers = undefined;
+        processor._colorTexture = undefined;
+        processor._ecAndLogDepthTexture = undefined;
+        processor._depthTexture = undefined;
+        processor._drawCommands = undefined;
+        processor._clearCommands = undefined;
+    }
+
+    function createFramebuffers(processor, context) {
+        var screenWidth = context.drawingBufferWidth;
+        var screenHeight = context.drawingBufferHeight;
+
+        var colorTexture = new Texture({
+            context : context,
+            width : screenWidth,
+            height : screenHeight,
+            pixelFormat : PixelFormat.RGBA,
+            // Firefox as of 57.02 throws FRAMEBUFFER_UNSUPPORTED 0x8CDD if this doesn't match what's in ecTexture
```

Comment:
Is there an issue to write a Firefox issue about this?  We don't have the time to create a stripped down test case for FF now.

Reply:
Opened an issue, but just realized we could pack the float to RGBA, which would also handily remove the dependence on `floatingPointTexture`.

Maybe this is still enough of an issue that we should be aware of it though? @lilleyse what do you think?

Reply:
Ah wait, from http://aras-p.info/blog/2009/07/30/encoding-floats-to-rgba-the-final/ it looks like our technique in `unpackDepth.glsl` is designed for range [0...1). So packing depth to RGBA isn't a good idea right now.

Reply:
Aheh. Also forgot that we'd need a much bigger buffer in terms of pixel count, since 1 RGBA -> 1 float.